src/main/java/com/internhub/data/positions/scrapers/strategies/impl/PositionBFSMTStrategy.java

```diff
@@ -0,0 +1,84 @@
+package com.internhub.data.positions.scrapers.strategies.impl;
+
+import com.google.common.collect.Lists;
+import com.internhub.data.models.Company;
+import com.internhub.data.models.Position;
+import com.internhub.data.pages.Page;
+import com.internhub.data.positions.extractors.PositionExtractor;
+import org.openqa.selenium.WebDriver;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.PriorityQueue;
+import java.util.Set;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+/**
+ * Had to do something like this to escape the schedule executor service task
+ */
+class ScheduledDelayFinishedException extends RuntimeException {
```

Comment:
If you have a better way to do this please tell me. I tried using a CountDownLatch initialized to max links, but then we waste time on companies that don't have that many links in their tree.